index.js

```diff
@@ -145,31 +144,42 @@ module.exports = class ServeDrive extends ReadyResource {
 
   async _onrequest (req, res) {
     if (req.method !== 'GET' && req.method !== 'HEAD') {
-      res.writeHead(400).end()
+      res.writeHead(400)
+      res.end()
       return
     }
 
-    const { pathname, searchParams } = new URL(req.url, 'http://localhost')
+    const { pathname, searchParams } = new URL(req.url, `http://${LOCALHOST}`)
     const version = searchParams.get('checkout')
     const id = searchParams.get('drive') // String or null
     const filename = decodeURI(pathname)
 
-    const drive = await this.getDrive(id, filename)
+    let drive = null
+    let error = null
 
     try {
+      drive = await this.getDrive(id, filename)
       await this._driveToRequest(drive, req, res, filename, id, version)
     } catch (e) {
       safetyCatch(e)
-      this.emit('request-error', e)
+      error = e
+    }
 
-      if (!res.headersSent) {
-        res.writeHead(500)
-        const msg = e.code || e.message
-        res.end(msg)
-      }
-    } finally {
-      await this.releaseDrive(id)
+    try {
+      if (drive !== null) await this.releaseDrive(id)
+    } catch (e) {
+      safetyCatch(e)
+      error = e
```

Comment:
That's overwriting the error variable. It should emit `request-error` on each catch so you don't lose any

Reply:
gets really complicated with reentrys, would just say thats a limitation that we are ok with

Reply:
If you want to emit at the end as per the other comment, then array of errors, but definitively not losing the first error as that will be more common probably

Reply:
why tho? super edge case

Reply:
(btw wrote my last comment before seeing yours)

Reply:
ie, request fails and drive closure fails. the only thing you can really do request-error is log anyway

Reply:
added a comment

Reply:
Ok but why the second is the important one? We always work with the first error

Reply:
first one is mostly just "connection closed" errors, this is actual user code. happy to change it to the former one.